internal/webhooks/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -109,10 +109,6 @@ func (webhook *BareMetalHost) validateChanges(oldObj *metal3api.BareMetalHost, n
 		errs = append(errs, errors.New("bootMACAddress can not be changed once it is set"))
 	}
 
-	if oldObj.Spec.ExternallyProvisioned != newObj.Spec.ExternallyProvisioned {
```

Comment:
This is a valid bug fix.
According to the state diagram, it is valid to remove the externallyProvisioned flag: https://github.com/metal3-io/baremetal-operator/blob/main/docs/baremetalhost-states.md
I have no idea why the webhook is preventing it.

Reply:
To echo what I said in the bug: if removing externallyProvisioned unconditionally deprovisions the host, we need to make sure that `image` and `customDeploy` are not set.

Or we need to permit them to be set and keep the host `Provisioned`.

Reply:
_Overall I'm wondering if the approach originally for the externallyProvisioned would have been to follow the cert-manager pattern and had a ProvisionRequest CR where some external system could fulfill the provision instead of hacking around the BareMetalHost CR._


Just to reflect on this point, this would be a nice approach but in practice the system that has done the provisioning might not be cloud native or not even automated at all so a ProvisionRequest CR would provide some optional extra functionality, but it would still not cover all the use cases.

Reply:
> Overall I'm wondering if the approach originally for the externallyProvisioned would have been to follow the cert-manager pattern and had a ProvisionRequest CR where some external system could fulfill the provision instead of hacking around the BareMetalHost CR.

Well, we should have had a separate CRD for provisioning for sure. The HostClaim work https://github.com/metal3-io/metal3-docs/pull/408 is a step in that direction but it does not address externally provisioned.

> Just to confirm (since I was not at the working group meetings) the desired behavior is

I think your summary is correct, except that: "If the `Image` **or** `customDeploy` are set" (one is enough).